src/recall.cpp

```diff
@@ -426,7 +423,9 @@ static void memoryMagicSkills(uint32_t memory_spell_flags, uint32_t monster_spel
 
 // Do we know how hard they are to kill? Armor class, hit die.
 static void memoryKillDifficulty(Creature_t *creature, uint32_t monster_kills) {
-    if (!knowarmor(creature->level, monster_kills)) {
+    // the higher the level of the monster, the fewer the kills you need
+    // Original knowarmor macro inlined
+        if (monster_kills <= 304u / (4u + creature->level)) {
```

Comment:
I keep looking at this, yet I'm still not convinced that these unsigned castings are needed.

A literal `4` (an `int`) plus a `uint8_t` number will never overflow its `int` type, and the result will always be a positive number less than `304`, so dividing `304` by the result will also always be a positive number.

Casting these two literals to `unsigned` types brings no benefit that I can see - or have I missed something?